Seems like you need to:

```
Error: Dependencies in integration-tests/gradle/pom.xml are outdated!
Error: Run update-dependencies.sh in integration-tests/gradle and add the modified pom.xml file to your commit.
Error: Process completed with exit code 1.
```

:)

--- a/.github/workflows/ci-actions-incremental.yml
+++ b/.github/workflows/ci-actions-incremental.yml
@@ -103,6 +103,8 @@ jobs:
           then
             echo -e '\033[0;31mError:\033[0m Dependencies to extension artifacts are outdated!' 1>&2
             echo -e '\033[0;31mError:\033[0m Run ./update-extension-dependencies.sh and add the modified pom.xml files to your commit.' 1>&2
+            echo -e '\033[0;31mError:\033[0m Diff is:' 1>&2
+            git --no-pager diff '*pom.xml' 1>&2
             exit 1
           fi
       - name: Tar Maven Repo
@@ -407,6 +409,8 @@ jobs:
           then
             echo -e '\033[0;31mError:\033[0m Dependencies in integration-tests/gradle/pom.xml are outdated!' 1>&2
             echo -e '\033[0;31mError:\033[0m Run update-dependencies.sh in integration-tests/gradle and add the modified pom.xml file to your commit.' 1>&2
+            echo -e '\033[0;31mError:\033[0m Diff is:' 1>&2
+            git --no-pager diff '*pom.xml' 1>&2
             exit 1
           fi
         working-directory: integration-tests/gradle

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java
@@ -32,7 +32,7 @@ public class QuarkusGenerateCode extends QuarkusTask {
     public static final String QUARKUS_GENERATED_SOURCES = "quarkus-generated-sources";
     public static final String QUARKUS_TEST_GENERATED_SOURCES = "quarkus-test-generated-sources";
     // TODO dynamically load generation provider, or make them write code directly in quarkus-generated-sources
-    public static final String[] CODE_GENERATION_PROVIDER = new String[] { "grpc" };
+    public static final String[] CODE_GENERATION_PROVIDER = new String[] { "grpc", "avpr", "avsc" };
 
     public static final String INIT_AND_RUN = "initAndRun";
     private Set<Path> sourcesDirectories;

--- a/integration-tests/gradle/pom.xml
+++ b/integration-tests/gradle/pom.xml
@@ -41,6 +41,11 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-avro</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-bootstrap-core</artifactId>
             <version>${project.version}</version>
         </dependency>
@@ -97,6 +102,19 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arc-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-avro-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/AvroDevModeTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/AvroDevModeTest.java
@@ -1,0 +1,22 @@
+package io.quarkus.gradle.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+
+public class AvroDevModeTest extends QuarkusDevGradleTestBase {
+    @Override
+    protected String projectDirectoryName() {
+        return "avro-simple-project";
+    }
+
+    @Override
+    protected void testDevMode() throws Exception {
+        assertThat(getHttpResponse("/hello")).isEqualTo("MAIL,SMS");
+
+        replace("src/main/avro/hello.avpr",
+                ImmutableMap.of(" \"symbols\" : [\"MAIL\", \"SMS\"]", " \"symbols\" : [\"EMAIL\"]"));
+
+        assertUpdatedResponseContains("/hello", "EMAIL");
+    }
+}

--- a/integration-tests/gradle/src/test/resources/avro-simple-project/build.gradle
+++ b/integration-tests/gradle/src/test/resources/avro-simple-project/build.gradle
@@ -1,0 +1,30 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+}
+
+repositories {
+    // in case a custom local repo is configured we are going to use that instead of the default mavenLocal()
+    if (System.properties.containsKey('maven.repo.local')) {
+        maven {
+            url System.properties.get('maven.repo.local')
+        }
+    } else {
+        mavenLocal()
+    }
+    mavenCentral()
+    gradlePluginPortal()
+}
+
+dependencies {
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'io.quarkus:quarkus-resteasy'
+    implementation 'io.quarkus:quarkus-avro'
+}
+
+group 'org.acme'
+version '1.0.0-SNAPSHOT'
+
+test {
+    systemProperty "java.util.logging.manager", "org.jboss.logmanager.LogManager"
+}

--- a/integration-tests/gradle/src/test/resources/avro-simple-project/gradle.properties
+++ b/integration-tests/gradle/src/test/resources/avro-simple-project/gradle.properties
@@ -1,0 +1,2 @@
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformGroupId=io.quarkus

--- a/integration-tests/gradle/src/test/resources/avro-simple-project/settings.gradle
+++ b/integration-tests/gradle/src/test/resources/avro-simple-project/settings.gradle
@@ -1,0 +1,18 @@
+pluginManagement {
+    repositories {
+        // in case a custom local repo is configured we are going to use that instead of the default mavenLocal()
+        if (System.properties.containsKey('maven.repo.local')) {
+            maven {
+                url System.properties.get('maven.repo.local')
+            }
+        } else {
+            mavenLocal()
+        }
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    plugins {
+        id 'io.quarkus' version "${quarkusPluginVersion}"
+    }
+}
+rootProject.name = 'avro-project'

--- a/integration-tests/gradle/src/test/resources/avro-simple-project/src/main/avro/hello.avpr
+++ b/integration-tests/gradle/src/test/resources/avro-simple-project/src/main/avro/hello.avpr
@@ -1,0 +1,12 @@
+{
+  "protocol" : "HelloProtocol",
+  "namespace" : "org.acme.quarkus.hello",
+  "types" : [
+    {
+      "type": "enum",
+      "namespace": "org.acme.quarkus.hello",
+      "name": "Provider",
+      "symbols" : ["MAIL", "SMS"]
+    }
+  ]
+}

--- a/integration-tests/gradle/src/test/resources/avro-simple-project/src/main/java/org/acme/quarkus/sample/HelloResource.java
+++ b/integration-tests/gradle/src/test/resources/avro-simple-project/src/main/java/org/acme/quarkus/sample/HelloResource.java
@@ -1,0 +1,22 @@
+package org.acme.quarkus.sample;
+
+import static java.util.stream.Collectors.joining;
+
+import java.util.Arrays;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.acme.quarkus.hello.Provider;
+
+@Path("/hello")
+public class HelloResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String providerValues() {
+        return Arrays.stream(Provider.values()).map(String::valueOf).collect(joining(","));
+    }
+}

